event/consumer.go

```diff
@@ -2,77 +2,79 @@ package event
 
 import (
 	"context"
+	"fmt"
 
-	"github.com/ONSdigital/dp-cantabular-csv-exporter/config"
 	"github.com/ONSdigital/dp-cantabular-csv-exporter/schema"
 	kafka "github.com/ONSdigital/dp-kafka/v2"
-	"github.com/ONSdigital/log.go/log"
+	"github.com/ONSdigital/log.go/v2/log"
 )
 
-//go:generate moq -out mock/handler.go -pkg mock . Handler
-
-type Handler interface {
-	Handle(ctx context.Context, cfg *config.Config, instanceComplete *InstanceComplete) error
-}
-
 // Consume converts messages to event instances, and pass the event to the provided handler.
-func Consume(ctx context.Context, messageConsumer kafka.IConsumerGroup, handler Handler, cfg *config.Config) {
-
+func (p *Processor) Consume(ctx context.Context, cg kafka.IConsumerGroup, h Handler) {
 	// consume loop, to be executed by each worker
 	var consume = func(workerID int) {
 		for {
 			select {
-			case message, ok := <-messageConsumer.Channels().Upstream:
+			case msg, ok := <-cg.Channels().Upstream:
 				if !ok {
 					log.Event(ctx, "closing event consumer loop because upstream channel is closed", log.INFO, log.Data{"worker_id": workerID})
 					return
 				}
-				messageCtx := context.Background()
-				processMessage(messageCtx, message, handler, cfg)
-				message.Release()
-			case <-messageConsumer.Channels().Closer:
+
+				msgCtx, _ := context.WithCancel(ctx)
```

Comment:
The cancel function should be called, not discarded, to avoid a context leak

Reply:
You've been listening to that pesky linter again haven't you... :) AFAIK a context leak is only an issue when spawning new goroutines.. the leak being the goroutine never finishes. We can't put it in a defer statement here as the function doesn't end.. So in reality I don't think adding it will make any difference. Can do it for the sake of doing it though..